dynamodb-enhanced/deployment/src/main/java/io/quarkus/amazon/dynamodb/enhanced/deployment/DynamodbEnhancedProcessor.java

```diff
@@ -43,6 +66,58 @@ AdditionalBeanBuildItem producer() {
         return AdditionalBeanBuildItem.unremovableOf(DynamodbEnhancedClientProducer.class);
     }
 
+    @BuildStep
+    void setup(
+            BeanRegistrationPhaseBuildItem beanRegistrationPhase,
+            BuildProducer<RequireAmazonClientBuildItem> requireClientProducer) {
+
+        Optional<DotName> syncClassName = Optional.empty();
+        Optional<DotName> asyncClassName = Optional.empty();
+
+        //Discover all clients injections in order to determine if async or sync client is required
+        for (InjectionPointInfo injectionPoint : beanRegistrationPhase.getContext().get(BuildExtension.Key.INJECTION_POINTS)) {
+
+            org.jboss.jandex.Type injectedType = injectionPoint.getRequiredType();
+
+            if (DYNAMODB_ENHANCED_CLIENT.equals(injectedType.name())) {
+                syncClassName = Optional.of(DYNAMODB_CLIENT);
+            }
+            if (DYNAMODB_ENHANCED_ASYNC_CLIENT.equals(injectedType.name())) {
+                asyncClassName = Optional.of(DYNAMODB_ASYNC_CLIENT);
+            }
+        }
+
+        if (syncClassName.isPresent() || asyncClassName.isPresent()) {
+            requireClientProducer.produce(new RequireAmazonClientBuildItem(syncClassName, asyncClassName));
+        }
+    }
+
+    @BuildStep
+    @Record(ExecutionTime.STATIC_INIT)
```

Comment:
I think it's fine given what the methods recorded do.

Reply:
integration test passed and a user testing in real application reported it worked without regression, so should be safe